### PR TITLE
Updated and improved flow of contributing.md doc

### PR DESCRIPTION
**Pull request by zipesb**
Added notes or improved wording for
- yarn version
- installing windows build tools
- finding recommended extensions in VSCode
- only running a local copy versus contributing
- installation of Snore Toast for windows machines
- generating Bungie API credentials
- accessing the locally hosted dev server

And changed the order of the "Get your own API key" and "Start Dev Server" steps

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -13,62 +13,68 @@ Here are some tips to make sure your Pull Request (PR) can be merged smoothly:
...
 * Linux-based developers will need to install `build-essential` (`sudo apt-get install -y build-essential`) prior to running `yarn install`.
 
 **Docker Development**
 * As an alternative to installing the above, you can try a docker-based development environment, but be aware that none of the core developers use it so it may be broken. See: [Docker Development Guide](Docker.md)
 
-
 ### Clone the repo
 
-To locally **run a copy** of DIM, you can simply clone the code repository:
+If you only want to **run a copy** of DIM locally and are not interested in contributing changes to the project, you can directly clone the original repository to your local machine using
```
Comment (bhollis) [inline docs/CONTRIBUTING.md:34]: Not sure I understand the intent behind this change. You need to check out and run the app locally whether you want to contribute or not. You can choose to fork later if you want, and can switch the remote to point to your fork or add another remote.

**Review by bhollis — COMMENTED** [review]
Thanks for cleaning this up!

File: docs/CONTRIBUTING.md (hunk 1 above)
Comment (zipesb) [reply docs/CONTRIBUTING.md]: I've changed this line back to the original wording. My understanding of git workflows is still not perfect, thanks for explaining!

**Review by nev-r — APPROVED** [review]
really helpful, thank you

**Review by bhollis — APPROVED** [review]

**Pull request merged**